apps/etterlatte-vedtaksvurdering/src/main/kotlin/vedtaksvurdering/VedtakEtterbetaling.kt

```diff
@@ -0,0 +1,55 @@
+package no.nav.etterlatte.vedtaksvurdering
+
+import no.nav.etterlatte.libs.common.tidspunkt.norskKlokke
+import no.nav.etterlatte.libs.common.vedtak.Periode
+import no.nav.etterlatte.libs.common.vedtak.Utbetalingsperiode
+import java.time.Clock
+import java.time.Month
+import java.time.YearMonth
+
+val WAY_INTO_THE_FUTURE: YearMonth = YearMonth.of(2999, Month.DECEMBER)
+
+internal fun erVedtakMedEtterbetaling(
+    vedtakSomBehandles: Vedtak,
+    vedtaksvurderingRepository: VedtaksvurderingRepository,
+    clock: Clock = norskKlokke(),
+): Boolean {
+    when (val innhold = vedtakSomBehandles.innhold) {
+        is VedtakBehandlingInnhold -> {
+            val now = YearMonth.now(clock)
+
+            if (innhold.virkningstidspunkt < now) {
+                val ferdigstilteVedtak = vedtaksvurderingRepository.hentFerdigstilteVedtak(vedtakSomBehandles.soeker)
+                val tidligereVedtakTidslinje = Vedtakstidslinje(ferdigstilteVedtak).sammenstill(YearMonth.of(2024, Month.JANUARY))
+                val tidligereUtbetalingsperioder =
+                    tidligereVedtakTidslinje
+                        .filter { it.innhold is VedtakBehandlingInnhold }
+                        .flatMap { (it.innhold as VedtakBehandlingInnhold).utbetalingsperioder }
+                        .filter { it.periode.fom < now }
+
+                // Finn tidligere utbetalingsperiode som matcher ny(e)
+                // og sjekk om det er endring i beløp, hvis ny er høyere så er det etterbetaling
+                return innhold.utbetalingsperioder
```

Comment:
Kan man gjøre denne antagelsen når vi kommer på revurderinger eller har man da behov for å simulere for å kunne vite om det blir etterbetaling eller ikke? Fungerer vel uansett for førstegangsbehandling da vi ikke har noe tidligere vedtak.